### PR TITLE
Catch SSL ImportError

### LINKED ISSUE #204: Catch Lack of SSL module
We should catch this error and print an error message

```
linuxuser@linuxuser-VirtualBox /media/linuxuser/user/cs61a/lab/lab00 $ python3 ok --submit --debug
DEBUG  | ok.py:112 | Namespace(all=False, authenticate=False, backup=False, config=None, debug=True, export=False, get_token=False, hint=False, insecure=False, interactive=False, latest=False, local=False, lock=False, no_hints=False, no_update=False, question=None, restore=False, revise=False, score=False, score_out=<client.utils.output._OutputLogger object at 0x7f053f1f5f28>, server='ok.cs61a.org', style=False, submit=True, tests=False, timeout=10, unlock=False, update=False, verbose=False, version=False)
DEBUG  | ok.py:116 | Python 64bit
INFO  | assignment.py:53 | Loaded config from lab00.ok
=====================================================================
Assignment: Lab 0
OK, version v1.6.14
=====================================================================

INFO  | assignment.py:125 | Loading tests
INFO  | assignment.py:143 | Loaded lab00.py:twenty_sixteen
INFO  | assignment.py:201 | Loading protocols
INFO  | assignment.py:209 | Loaded protocol "restore"
INFO  | assignment.py:209 | Loaded protocol "file_contents"
INFO  | assignment.py:209 | Loaded protocol "analytics"
WARNING  | guidance.py:85 | Failed to read .ok_guidance file. It may not exist
DEBUG  | guidance.py:87 | Guidance loaded with status: False
INFO  | assignment.py:209 | Loaded protocol "unlock"
INFO  | assignment.py:209 | Loaded protocol "grading"
WARNING  | ok.py:156 | Assignment could not load
Traceback (most recent call last):
  File "ok/client/api/assignment.py", line 204, in _load_protocols
    module = importlib.import_module(self._PROTOCOL_PACKAGE + '.' + proto)
  File "/usr/local/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "ok/client/protocols/backup.py", line 10, in <module>
    import ssl
  File "/usr/local/lib/python3.5/ssl.py", line 98, in <module>
    import _ssl             # if we can't import it, let the error propagate
ImportError: No module named '_ssl'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "ok/client/cli/ok.py", line 140, in main
    assign = assignment.load_assignment(args.config, args)
  File "ok/client/api/assignment.py", line 24, in load_assignment
    return Assignment(cmd_args, **config)
  File "ok/client/sources/common/core.py", line 187, in __call__
    obj.post_instantiation()
  File "ok/client/api/assignment.py", line 119, in post_instantiation
    self._load_protocols()
  File "ok/client/api/assignment.py", line 206, in _load_protocols
    raise ex.LoadingException('Invalid protocol: {}'.format(proto))
client.exceptions.LoadingException: Invalid protocol: backup
Error loading assignment: Invalid protocol: backup
Checking for software updates...
INFO  | software_update.py:20 | Existing OK version: v1.6.14
INFO  | software_update.py:21 | Checking latest version from https://ok.cs61a.org/api/v3/version/ok-client
Network error when checking for updates.
WARNING  | software_update.py:29 | Network error when checking version from https://ok.cs61a.org/api/v3/version/ok-client: <urlopen error unknown url type: https>
Stack (most recent call last):
  File "/usr/local/lib/python3.5/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "ok/__main__.py", line 9, in <module>
    ok.main()
  File "ok/client/cli/ok.py", line 170, in main
    client.FILE_NAME)
  File "ok/client/utils/software_update.py", line 29, in check_version
    str(e), stack_info=True)
```

While we are at it - we should also log detailed python versions

### PR DESCRIPTION
**Pull request by colinschoen**
Not sure of a clean way to unit test this without stubbing `import`. 

Resolves #204

### CODE REVIEW

**Comment by knrafto** [comment]
Were you able to test manually?

This will not allow the student to do anything if they haven't installed SSL. They should be able to do work and submit with `--insecure` though.

**Comment by Sumukh** [comment]
There's already a SSL check function in `auth.py`. Maybe move that into utils and use that in both places?

File: client/utils/auth.py [hunk 1]
```diff
@@ -18,6 +18,8 @@
                                  create_config_directory)
 from client.utils.html import (auth_html, partial_course_html,
                                partial_nocourse_html, red_css)
+from client.utils import network
+from client.utils.sanction import Client
```
Comment (Sumukh) [inline client/utils/auth.py]: This is no longer correct (see `from sanction import Client` above) (see master/#209)

Comment (colinschoen) [inline client/utils/auth.py]: Done, I missed that change. Thanks.

**Comment by colinschoen** [comment]
Don't merge yet. I need to do some manual testing after these changes.

**Comment by knrafto** [comment]
@colinschoen what's the status of this PR?

**Comment by colinschoen** [comment]
Whoops, forgot this one was hanging. Will test by Sunday.

**Comment by colinschoen** [comment]
I'm content with these changes, but compiling python without SSL is seemingly trickier than I thought. Do we have anyone (staff or student) that has this issue we could test on?

**Comment by knrafto** [comment]
Finally getting around to this. Thanks Colin!

**Comment by colinschoen** [comment]
Thanks for reviewing.

**Pull request merged**